include/logging/log.h

```diff
@@ -267,7 +271,11 @@ int log_printk(const char *fmt, va_list ap);
  *	   allocated. String may be truncated if input string does not fit in
  *	   a buffer from the pool (see CONFIG_LOG_STRDUP_MAX_STRING).
  */
+#if defined(CONFIG_LOG)
 char *log_strdup(const char *str);
+#else
+#define log_strdup(...) ""
```

Comment:
This looks wrong to me, `log_strdup()` will allocate the string using a memory allocator and allocates memory in a pool. The way you are doing if someone call `log_free()` will crash.

Reply:
In this case log_strdup() and log_free() don't actually exist since CONFIG_LOG is not set.  The issue is that log_strdupI() is often used within sprintk calls so you can't define it as a semicolon or expand to nothing, that causes a compile error.   The safest way I could come up with is to equate it to an empty string which doesn't hurt terribly.

Reply:
actually log_free will check the memory is alloced by log_strdup or not.